### PR TITLE
Wrong version of GSL

### PR DESCRIPTION
**Pull request by oleg-alexandrov**
Based on the file environment.yml in this repository, the GSL version used is 2.2.1. Yet CMakeLists.txt insists on version 19. That is wrong. The latest stable release of GSL is 2.5. Maybe there 1.9 was meant. In either case, apparently this version number is ignored anyway, the only reason I found out is when I switched from your homegrown isis/cmake/FindGSL.cmake when does not use the version in its logic to miniconda3/envs/isis3/share/cmake-3.11/Modules/FindGSL.cmake which does.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jlaura** [comment]
@Kelvinrr Can you give this a review early next week and then we can merge. Perfect timing inside an RC so we can get this change propagated with the 3.8.0 release. @oleg-alexandrov Thanks!

**Comment by jessemapel** [comment]
Is there a reason why we use our own find* scripts and not the default cmake ones? Looking at histories, most of the changes are just updating messages.

**Comment by Kelvinrr** [comment]
looks like the 19 was a relic from when CmakeLists.txt still looked at v007. Not sure why we are not using the CMake script, but, in remembering when this was starting out, I'm pretty sure the original author put in 19 because that was embedded into the library path similar to how versions are normally used for C++ libraries and mistaken number that for the GSL version instead of looking at the gsl-config output. 

``` bash
(isis) igswzawglt0091:~ kelvin$ ls /Users/kelvin/anaconda3/envs/isis/lib/libgsl*
/Users/kelvin/anaconda3/envs/isis/lib/libgsl.19.dylib     /Users/kelvin/anaconda3/envs/isis/lib/libgsl.la
/Users/kelvin/anaconda3/envs/isis/lib/libgsl.a            /Users/kelvin/anaconda3/envs/isis/lib/libgslcblas.0.dylib
/Users/kelvin/anaconda3/envs/isis/lib/libgsl.dylib        /Users/kelvin/anaconda3/envs/isis/lib/libgslcblas.dylib
(isis) igswzawglt0091:~ kelvin$ conda list gsl
# packages in environment at /Users/kelvin/anaconda3/envs/isis:
#
# Name                    Version                   Build  Channel
gsl                       2.2.1           blas_openblas_3  [blas_openblas]  conda-forge
```

TLDR: we should remove our FindGSL. I'll approve this and make an issue to remove Find scripts already provided by Cmake, I doubt this is the only one.

**Review by Kelvinrr — APPROVED** [review]
